vowpalwabbit/cb_adf.cc

```diff
@@ -543,21 +549,24 @@ base_learner* cb_adf_setup(options_i& options, vw& all)
 
   if (options.was_supplied("baseline") && check_baseline_enabled) { options.insert("check_enabled", ""); }
 
-  auto ld = scoped_calloc_or_throw<cb_adf>(all.sd, cb_type, &all.model_file_ver, rank_all, clip_p, no_predict);
+  auto ld = VW::make_unique<cb_adf>(all.sd, cb_type, &all.model_file_ver, rank_all, clip_p, no_predict);
 
   auto base = as_multiline(setup_base(options, all));
   all.example_parser->lbl_parser = CB::cb_label;
 
   cb_adf* bare = ld.get();
-
-  learner<cb_adf, multi_ex>& l = init_learner(ld, base, learn, predict, problem_multiplier,
-      prediction_type_t::action_scores, all.get_setupfn_name(cb_adf_setup), ld->learn_returns_prediction());
-
-  l.set_finish_example(CB_ADF::finish_multiline_example);
-  l.set_print_example(CB_ADF::update_and_output);
+  bool lrp = ld->learn_returns_prediction();
+  auto* l = make_reduction_learner(std::move(ld), base, learn, predict, all.get_setupfn_name(cb_adf_setup))
+                .set_learn_returns_prediction(lrp)
+                .set_params_per_weight(problem_multiplier)
+                .set_prediction_type(prediction_type_t::action_probs)
```

Comment:
Looks like this one should be action_scores

```suggestion
                .set_prediction_type(prediction_type_t::action_scores)
```